bin.js

```diff
@@ -121,26 +122,46 @@ yargs.command('run <system> [location] [test]', 'Run your package\'s tests in re
     // Add the raw package.json template
     const rnPkg = require(`${root}/template-package.json`)
     await writeFile(`${root}/package.json`, JSON.stringify(rnPkg, null, 2) + '\n')
+
+    // Install any dependencies of the wrapper app
+    await logExec('npm', ['i'], { cwd: root, quiet: !verbose })
+
+    console.log('## react-native:npm pre-pack')
+
+    // Merge devDependencies with dependencies
+    const { dependencies = {}, devDependencies = {} } = packageJSON
+    const combinedDeps = { ...dependencies, ...devDependencies }
+    const modifiedPackageJSON = { ...packageJSON, dependencies: combinedDeps }
+    await writeFile(packageJSONLocation, JSON.stringify(modifiedPackageJSON))
+
+    console.log('## react-native:npm pack')
+
     try {
-      // Install any dependencies of the wrapper app
-      await logExec('npm', ['i'], { cwd: root, quiet: !verbose })
+    // Pack up package
+      await logExec('npm', ['pack'], {
+        cwd: packageLocation,
+        quiet: !verbose
+      })
     } finally {
-      // Add the package name to dependencies so it will compile
-      rnPkg.dependencies[packageJSON.name] = ''
-      await writeFile(`${root}/package.json`, JSON.stringify(rnPkg, null, 2) + '\n')
+    // Restore old package.json
+      await writeFile(packageJSONLocation, packageRaw)
     }
 
-    // Make sure the folder exists for copying over the package
-    console.log(`## react-native:preparing-dep [target=${target}]`)
-    await mkdirp(path.dirname(target))
+    // Get a reference to the tar file
+    const tarName = `${packageJSON.name}-${packageJSON.version}.tgz`
+    const tarPath = path.join(packageLocation, tarName)
 
-    // Copy over the package contents
-    console.log(`## react-native:copy-build [src=${packageLocation} target=${target}]`)
+    try {
+      console.log('## react-native:npm install .tgz')
 
-    await cpr(packageLocation, target, {
-      // Avoid recursive copying
-      filter: (item) => !item.startsWith(`${__dirname}/`)
-    })
+      // Install from pack file
+      await logExec('npm', ['i', tarPath], {
```

Comment:
When reviewing this I was wondering if it shouldn't be `--save`: meaning that the package should be saved in the package.json of the project in case some steps down the line don't do that properly. But there is a whole rat-tail of questions following this, so let's leave it like this for now 😉

Reply:
Yeah... I guess we'll deal with it if it's a problem in the future? 😅